packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -52,8 +50,10 @@ export const ColorScaleBlock: FC<BlockProps> = ({ appBridge }) => {
     const resizeStartPos = useRef<Nullable<number>>();
     const resizeStartWidth = useRef<Nullable<number>>();
     const lastDragPos = useRef<Nullable<number>>();
+    const [addColorDisabled, setAddColorDisabled] = useState(false);
     const positionWhereSiblingColorNeededResizing = useRef<number>(0);
     const originalSiblingColorWidthBeforeResizing = useRef<number>(0);
+    const [draggedColorWidth, setDraggedColorWidth] = useState<Nullable<number>>(null);
```

Comment:
I would move this useState into `ColorSquare`. As you only need it there you don't need to define it here and pass it down

Reply:
The reason why it can't be moved in there is because we are passing the width of the dragged block over to other blocks, who don't have access to that info within ColorSquare.

Reply:
When a color is dragged, the width needs to be passed to sibling colors so that a dropzone with a custom width can be displayed.  

There is no way for the sibling colors to know the width of the dragged color unless it is stored in ColorScaleBlock.tsx.  Maybe we can have a quick call so I can show you what I mean.

Reply:
I just saw [Gorans comment](https://github.com/Frontify/guideline-blocks/pull/440/files/6554abc520bc95f8ddbcd52af75f87f750b0b84b#r1066816170). If you do this. Then I think you don't need to pass the `setColorWidth`